README.md

```diff
@@ -140,24 +140,36 @@ kubectl apply -f https://raw.githubusercontent.com/fluent/fluentbit-operator/mas
 The Fluent Bit section of the Fluent Operator supports different CRI `docker`, `containerd`,  and `CRI-O`. 
 `containerd` and `CRI-O` use the `CRI Log` format which is different with `docker`, they requires additional parser to parse JSON application logs. You should set different `containerRuntime` depending on your container runtime.
 
-If your container runtime is `docker`
+If your container runtime is `docker`:
```

Comment:
A question on this: is it purely to handle the right parser? Seems like we should default to using any/all of them so it always works but people who know what they're doing can say "use this particular parser".

Reply:
The docs for example seem to do this: https://docs.fluentbit.io/manual/pipeline/inputs/tail#multiline-and-containers-v1.8

Reply:
Why can't we just use all of them like the docs do? Having to specify the specific runtime you are using seems brittle and easy to fail on, particularly with mixed nodes. I think it would be better to say the default is we'll try all parsers but if you know what you're doing you can specify a particular one - can you also customise it this way?

Reply:
Yes, that would make sense. But I found the deployment is different according the given Runtime field:

See belows:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: fluent-operator
  labels:
    app.kubernetes.io/component: operator
    app.kubernetes.io/name: fluent-operator
spec:
  selector:
    matchLabels:
      app.kubernetes.io/component: operator
      app.kubernetes.io/name: fluent-operator
  template:
    metadata:
      labels:
        app.kubernetes.io/component: operator
        app.kubernetes.io/name: fluent-operator
    spec:
      {{- if eq .Values.containerRuntime "docker" }}
      volumes:
      - name: env
        emptyDir: {}
      - name: dockersock
        hostPath:
          path: /var/run/docker.sock
      initContainers:
      - name: setenv
        image: {{ .Values.operator.initcontainer.repository }}:{{ .Values.operator.initcontainer.tag }}
        command:
        - /bin/sh
        - '-c'
        - set -ex;
          echo CONTAINER_ROOT_DIR=$(docker info -f '{{`{{.DockerRootDir}}`}}' 2> /dev/null) > /fluent-operator/fluent-bit.env
        volumeMounts:
        - name: env
          mountPath: /fluent-operator
        - name: dockersock
          readOnly: true
          mountPath: /var/run/docker.sock
      {{- else if eq .Values.containerRuntime "containerd" }}
      volumes:
      - name: env
        emptyDir: {}
      initContainers:
      - name: setenv
        image: {{ .Values.operator.initcontainer.repository }}:{{ .Values.operator.initcontainer.tag }}
        command:
        - /bin/sh
        - '-c'
        - set -ex;
          echo CONTAINER_ROOT_DIR={{ .Values.operator.logPath.containerd }} > /fluent-operator/fluent-bit.env
        volumeMounts:
        - name: env
          mountPath: /fluent-operator
      {{- else if eq .Values.containerRuntime "crio" }}
      volumes:
        - name: env
          emptyDir: {}
      initContainers:
      - name: setenv
        image: {{ .Values.operator.initcontainer.repository }}:{{ .Values.operator.initcontainer.tag }}
        command:
        - /bin/sh
        - '-c'
        - set -ex;
          echo CONTAINER_ROOT_DIR={{ .Values.operator.logPath.crio }} > /fluent-operator/fluent-bit.env
        volumeMounts:
        - name: env
          mountPath: /fluentbit-operator
      {{- end }}
      containers:
      - name: fluent-operator
        image: {{ .Values.operator.container.repository }}:{{ .Values.operator.container.tag }}
        resources:
          {{- toYaml .Values.operator.resources | nindent 10 }}
        env:
          - name: NAMESPACE
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: metadata.namespace
        volumeMounts:
        - name: env
          mountPath: /fluent-operator
      serviceAccountName: fluent-operator
      {{- if .Values.operator.imagePullSecrets }}
      imagePullSecrets:
      {{ toYaml .Values.operator.imagePullSecrets | indent 8 }}
      {{- end }}
```

Reply:
Any better suggestions to optimize this? /cc  @patrick-stephens @wenchajun @benjaminhuo @wanjunlei

Reply:
That's a lot of YAML and sounds very confusing: can you simplify it down to the generated per option?
From what I can tell it is down to different mount paths although then the docker option also has the socket exposed (which I'm not sure about and sounds like a potential security thing to document too) - why do we need this code duplication? 
However, probably not an issue for this PR so happy for it to proceed.

Reply:
We'll find out if there is an easier way to set the path. @wenchajun @wanjunlei @zhu733756 
Maybe just a field in FluentBit CR called `ContainerRuntime`, and the operator handles the rest?

Reply:
Deployment fluent-operator acts as controller to reconcile other resources, including the operator,  maybe cannot reconcile itself?

Reply:
Maybe in a different way, these path settings are to be passed to fluent bit I think, it's the fluent bit that actually read logs on these paths.